transform-plugins/docs/Flatten-transform.md

```diff
@@ -0,0 +1,37 @@
+# Flatten Transform
+
+## Description
+Flatten is a transform plugin that flattens nested data structures.
+
+## Use Case
+
+The transform is used  to convert a nested data structure  into a single flattened record where each key in the record 
+is a ```_``` name path to the node in the nested structure. 
+
+Configuration
+-------------
+**Fields to flatten:** Specifies the list of fields in the input schema to be flattened.
+
+**Level to limit flattening:** Limit flattening to a certain level in nested structures.
+
+**Prefix:** An optional prefix to be used while generating names of the flattened fields. This can be used to fix conflicts that can occur if fields have the same name after flattening. The prefix is added as <prefix>_<flattened_name >.
```

Comment:
What happens if there are records of record? What prefix will be used?

Reply:
@CuriousVini The prefix is used only if after flattening there are fields with same name, the prefix will be added to duplicate.

example:

` {
   "Address_Number": 100,
   "Address":{
   		"Number":100
   }
 }`

The output fields will be:
Address_Number, Prefix_Address_Number

Reply:
What would happen if there are 3 record fields in a schema with `Number` field?

Reply:
If there are more than one duplicate the value of last duplicate will be mapped to last prefix.
 In order to prevent this scenario we can do one of the following:
      1. Raise Exception during schema generation 
      2. Add counter to prefix: prefix_1_field, prefix_2_field ....

Reply:
For duplicate columns, This is the behavior we should have:
1. You get a validation error when validate button is pressed.
2. The error has a clear call to action. "Rename x, y, or z fields before using the flatten transform"